spectractor/extractor/spectrum.py

```diff
@@ -654,107 +749,204 @@ def load_spectrum(self, input_file_name, spectrogram_file_name_override=None,
+        self.my_logger.info(f'\n\tLoading disperser {self.disperser_label}...')
+        self.disperser = Hologram(self.disperser_label, D=parameters.DISTANCE2CCD,
+                                  data_dir=parameters.DISPERSER_DIR, verbose=parameters.VERBOSE)
+        self.my_logger.info(f'\n\tSpectrum loaded from {input_file_name}')
+        if parameters.OBS_OBJECT_TYPE == "STAR":
+            self.adr_params = [self.dec, self.hour_angle, self.temperature,
+                               self.pressure, self.humidity, self.airmass]
+
+        self.psf = load_PSF(psf_type=parameters.PSF_TYPE, target=self.target)
+        self.chromatic_psf = ChromaticPSF(self.psf, self.spectrogram_Nx, self.spectrogram_Ny,
+                                          x0=self.spectrogram_x0, y0=self.spectrogram_y0,
+                                          deg=self.spectrogram_deg, saturation=self.spectrogram_saturation)
+        if 'PSF_REG' in self.header and float(self.header["PSF_REG"]) > 0:
+            self.chromatic_psf.opt_reg = float(self.header["PSF_REG"])
+
+        # original, hard-coded spectrogram/table relative paths
+        spectrogram_file_name = input_file_name.replace('spectrum', 'spectrogram')
+        psf_file_name = input_file_name.replace('spectrum.fits', 'table.csv')
+        if spectrogram_file_name_override and psf_file_name_override:
+            self.fast_load = False
+            spectrogram_file_name = spectrogram_file_name_override
+            psf_file_name = psf_file_name_override
+            self.my_logger.info(f'\n\tApplying spectrogram filename override {spectrogram_file_name}')
+            self.my_logger.info(f'\n\tApplying psf filename override {psf_file_name}')
+
+        if not self.fast_load:
+            hdu_list = fits.open(input_file_name)
+            # load other spectrum info
+            if len(hdu_list) > 1:
+                self.cov_matrix = hdu_list["SPEC_COV"].data
+                if len(hdu_list) > 2:
+                    _, self.data_order2, self.err_order2 = hdu_list["ORDER2"].data
+                    if len(hdu_list) > 3:
+                        self.target.image = hdu_list["ORDER0"].data
+                        self.target.image_x0 = float(hdu_list["ORDER0"].header["IM_X0"])
+                        self.target.image_y0 = float(hdu_list["ORDER0"].header["IM_Y0"])
+            # load spectrogram info
+            if len(hdu_list) > 4:
+                self.spectrogram = hdu_list["S_DATA"].data
+                self.spectrogram_err = hdu_list["S_ERR"].data
+                self.spectrogram_bgd = hdu_list["S_BGD"].data
+                if len(hdu_list) > 7:
+                    self.spectrogram_bgd_rms = hdu_list["S_BGD_ER"].data
+                    self.spectrogram_fit = hdu_list["S_FIT"].data
+                    self.spectrogram_residuals = hdu_list["S_RES"].data
+            elif os.path.isfile(spectrogram_file_name):
```

Comment:
So `spectrogram_file_name` is ignored if the data is found inside the file. It makes sense, but shouldn't it therefore be a raise or at least a warning if you have both? I just think I'd have structured it a little different to make it more robust.

Reply:
It's the old load function and I don't know how to build it better, but I do understand your point. I think I will keep it like this.